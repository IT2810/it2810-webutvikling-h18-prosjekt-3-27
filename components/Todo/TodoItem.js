import {StyleSheet, Text, View, TextInput} from "react-native";
import Icon from "react-native-vector-icons/Ionicons";
import React, {Component} from "react";

export default class TodoItem extends Component {

  _handlePress = () => {
    this.props.toggleComplete(this.props.index);
  };

  _handleTextEdit = () => {
    this.props.onEditStart(this.props.item.key)
  };

  _handleTextChange = () => {

  };

  render() {
    let item = this.props.item;
    console.log(this.props.selected);
    console.log(this.props.item.key);
    return (
      <View>
        <View style={styles.itemContainer}>
          {(this.props.selected === this.props.item.key) ?
          <TextInput
            value={this.props.item.text}
            onChangeText={this._handleTextChange}
          />
         :
          <Text onPress={this._handleTextEdit} style={item.completed ? styles.itemCompleted : styles.item}>
            {item.text}
          </Text>

          }
          <Icon style={styles.button} name={item.completed ? "ios-checkmark-circle": "ios-radio-button-off"}  color={item.color} size={50}
                onPress={this._handlePress}/>
        </View>
        <View style={styles.hr}/>
      </View>
    );
  }
}
const styles = StyleSheet.create({
  itemContainer: {
    flexDirection: "row",
    alignItems: "center",
    justifyContent: "space-between",
  },
  item: {
    flex: 1,
    flexWrap: "wrap",
    paddingLeft: 10,
    paddingTop: 2,
    paddingBottom: 2,
    fontSize: 18,
    margin: 5,
  },
  itemCompleted: {
    flex: 1,
    flexWrap: "wrap",
    textDecorationLine: "line-through",
    color: "#D3D3D3",
    paddingLeft: 10,
    paddingTop: 2,
    paddingBottom: 2,
    fontSize: 18,
    margin: 5,
  },
  button: {
    marginRight: 5,
  },
  hr: {
    marginTop: 5,
    marginBottom: 5,
    height: 1,
    backgroundColor: "gray"
  },
});