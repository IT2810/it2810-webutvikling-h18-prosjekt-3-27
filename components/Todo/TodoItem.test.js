import React from 'react';
import renderer from 'react-test-renderer';

import TodoItem from "./TodoItem";


test('renders correctly', () => {
  const tree = renderer.create(<TodoItem
    item={{key: "key", text: "sometext", completed: false, color: "green"}}
  />).toJSON();
  expect(tree).toMatchSnapshot();
});
