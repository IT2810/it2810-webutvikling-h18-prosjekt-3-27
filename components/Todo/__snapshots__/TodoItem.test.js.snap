// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "space-between",
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      onPress={[Function]}
      style={
        Object {
          "flex": 1,
          "flexWrap": "wrap",
          "fontSize": 18,
          "margin": 5,
          "paddingBottom": 2,
          "paddingLeft": 10,
          "paddingTop": 2,
        }
      }
    >
      sometext
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
    />
  </View>
  <View
    style={
      Object {
        "backgroundColor": "gray",
        "height": 1,
        "marginBottom": 5,
        "marginTop": 5,
      }
    }
  />
</View>
`;
